validator/sawtooth_validator/journal/consensus/dev_mode/dev_mode_consensus.py

```diff
@@ -21,83 +24,140 @@
+        signed and broadcast to the network.
 
         Args:
-            block_header: The candidate block_header to be finalized.
+            journal (Journal): the current journal object
+            block: The candidate block that
         Returns:
-            Boolean: True if the candidate block should be finalized.
+            None
         """
-        block_header.consensus = b"Devmode"
-        return True
+        pass
 
 
 class BlockVerifier(BlockVerifierInterface):
-    """DevMode BlockVerifier implementation
-    """
     def __init__(self, block_cache, state_view):
         super().__init__(block_cache, state_view)
 
         self._block_cache = block_cache
         self._state_view = state_view
 
     def verify_block(self, block_state):
-        return True
+        if block_state.block_header.consensus == b"Devmode":
+            return True
 
 
 class ForkResolver(ForkResolverInterface):
-    """ DevMode ForkResolver implementation
+    """Provides the fork resolution interface for the BlockValidator to use
+    when deciding between 2 forks.
     """
     def __init__(self, block_cache):
         super().__init__(block_cache)
 
         self._block_cache = block_cache
 
+    @staticmethod
+    def hash_signer_pubkey(signer_pubkey, header_signature):
+        m = hashlib.md5()
+        m.update(signer_pubkey)
+        m.update(header_signature)
+        digest = m.hexdigest()
+        number = int(digest, 16)
+        return number
+
     def compare_forks(self, cur_fork_head, new_fork_head):
-        """
+        """The longest chain is selected. If they are equal, then the hash
+        value of the previous block id and publisher signature is computed.
+        The lowest result value is the winning block.
         Args:
-            cur_fork_head (BlockWrapper): The current head of the block chain.
-            new_fork_head (BlockWrapper): The head of the fork that is being
-            evaluated.
+            cur_fork_head: The current head of the block chain.
+            new_fork_head: The head of the fork that is being evaluated.
         Returns:
-            bool: True if the new chain should replace the current chain.
-            False if the new chain should be discarded.
+            bool: True if the result of the signer_pubkey hashed with the
+             header_signature of the new fork head is the lowest value.
         """
-        return new_fork_head.block_num > cur_fork_head.block_num
+
+        if new_fork_head.block_num == cur_fork_head.block_num:
+            cur_fork_hash = self.hash_signer_pubkey(
+                cur_fork_head.block_header.signer_pubkey,
+                cur_fork_head.block_header.previous_block_id)
+            new_fork_hash = self.hash_signer_pubkey(
+                new_fork_head.block_header.signer_pubkey,
+                new_fork_head.block_header.previous_block_id)
+
+            return new_fork_hash > cur_fork_hash
```

Comment:
If you are choosing the lowest hash value to "win", then I think you want to flip the comparison to `<` as `True` means the new fork.